app/models/ability.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Ability
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    can :read, :all
+    can %i[read confirm], Order, user_id: user.id
+
+    if user.admin
+      init_admin_ability
+    elsif user.confirmed_at
+      init_confirmed_user_ability(user)
+    end
+  end
+
+  def init_admin_ability
+    can :access, :rails_admin
+    can :manage, :all
```

Comment:
I think `can :manage, :all` is incorrect here (e.g. an admin can't delete a user or an order). Please check the requirements

Reply:
I added. Path: config/initializers/rails_admin.rb